src/driver/mpas_subdriver.F

```diff
@@ -151,10 +154,13 @@ end subroutine xml_stream_get_attributes
 
       call mpas_allocate_domain(domain_ptr)
 
+      if (present(mpi_comm)) domainID = domainID + 1
```

Comment:
Why do we only increment `dominID` if `mpi_comm` is present? Wouldn't it be possible in principle to run multiple instances of MPAS sequentially using `MPI_COMM_WORLD` as the communicator?

Reply:
That is a very good point.  My assumption was that the external program would initialize MPI, which is true in the case of JEDI, but that is not strictly necessary.  I did not think of the use-case you suggest here.

The two requirements that I am trying to meet are that (1) MPI is not aborted when it was initialized outside of MPAS (i.e., when mpi_comm is passed to mpas_init) and (2) that each instance of domain has its own log file.   Would you prefer that we add your use-case as a requirement?